### DISCUSSION TITLE
Are 'weekly' Meltano releases too frequent?

### DISCUSSION THREAD

**Discussion by aaronsteers** [meltano/meltano#6475] [Polls]
Our philosophy at Meltano has always been to release features to our users as soon as they are ready. This means we often have a 3 or 4 releases every month, which some users may have a hard time keeping up with.

Should we move to monthly releases, with only patch/hotfix updates in between our monthly cadence? 

And if we do move to a monthly release cadence, would it be helpful to have a separate "Meltano Insiders Build" so that early adopters can still start testing features as soon as they are ready?

_Sound off below!_

**Comment by pandemicsyn** [comment]
> 🚀 Same as today! Release as soon as the features are ready!

Just wanna point out that what we do today is slight different than "release as soon as features are ready" which sorta implies a "release on merge" pattern.  Our weekly release cadence is fast enough though that practically speaking it's basically the same.

**Comment by edgarrmondragon** [comment]
Changing our release strategy will most likely force us to change our git branching strategy, which is fine but requires work and has a slight learning curve for contributors (internal and external).

**Reply by pandemicsyn** [reply]
Not sure what @aaronsteers had in mind for a "insiders" build, but might just mean doing our release party once a month instead of weekly, and just doing nightly automated builds for previews (pushed to Gitlab python package registry? :trollface: ).

**Reply by edgarrmondragon** [reply]
@pandemicsyn I hadn't thought of it, but we can now do PyPI prereleases very easily, which `pip` doesn't install unless the user explicitly requests. Would that work as an "insiders" build if we make them often enough?

**Reply by pandemicsyn** [reply]
@edgarrmondragon - IMHO generally "insider" builds aren't available through regular distro channels anyway and usually require some form of explicit opt-in to obtain. So doing pypi prereleases seem's like a good approach.
